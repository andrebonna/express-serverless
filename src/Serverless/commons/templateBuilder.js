import React from 'react';
import ReactDOMServer from 'react-dom/server';
import ejs from 'ejs';
import Head from '../../View/containers/Head';
import App from '../../View/containers/App';

function headBuilder(title, metas) {

    const cssBundle = `${process.env.CDN_PATH || ''}bundle.css`;
    const head = React.createElement(Head, {
        title,
        metas,
        cssBundle
    });
    return ReactDOMServer.renderToString(head);
}

function contentBuilder(props) {
    return ReactDOMServer.renderToString(React.createElement(App, props));
}

export default function templateBuilder({ title, metas, props }, callback) {
    const data = {
        cdn_path: process.env.CDN_PATH || '',
        head: headBuilder(title, metas),
        content: contentBuilder(props)
    };
    ejs.renderFile(`${__dirname}/../../View/template.ejs`, data, callback);
}