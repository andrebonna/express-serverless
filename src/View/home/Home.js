import React, { Component } from 'react';
import './home.scss';

import ImageGallery from '../components/ImageGallery';

const IMAGES = [
    'http://wallpaperhey.com/Images/nature-image-for-iphone-Free-High-Resolution-Images.jpg',
    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSFjJ6Jk2PQNu-ZRBQM2FkeiZFEwx8R5AifYX6OPe6kPjMsDbze',
    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQirrr799v2o08VIVZsWAlEL66GkPlCZ5cJNja2GrN1cTGAkC6s',
    'https://i.pinimg.com/736x/5a/e9/50/5ae9501fc3b49810db7901873f77d6f7--beautiful-nature-photos-beautiful-days.jpg',
    'https://i.ytimg.com/vi/kIS5SRuMg3c/hqdefault.jpg',
    'https://i.ytimg.com/vi/x30YOmfeVTE/maxresdefault.jpg',
    'https://images.pexels.com/photos/129441/pexels-photo-129441.jpeg?w=1260&h=750&auto=compress&cs=tinysrgb',
    'http://images.all-free-download.com/images/graphiclarge/trail_nature_landscape_214997.jpg',
    'http://sim01.in.com/4fc598f2c9f2c0cdc5e0decc188d8d10_ft_xl.jpg'
];

export default class Home extends Component {
    render() {
        return <ImageGallery images={IMAGES} />;
    }
}