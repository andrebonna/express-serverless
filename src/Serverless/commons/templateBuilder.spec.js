import React from 'react';
import templateBuilder from './templateBuilder';
import Home from '../../View/home/Home';

describe('Test template Builder', () => {
    it('templateBuilder with options', (done) => {
        templateBuilder({
            title: 'titleMock',
            metas: {
                metaMock: 'metaContentMock'
            },
            props: {
                categories: [],
                children: <Home images={[]} />
            }
        }, (err, content) => {
            expect(content).toMatchSnapshot();
            done();
        });
    });
});