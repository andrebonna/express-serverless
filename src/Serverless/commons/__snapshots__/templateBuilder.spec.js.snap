// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test template Builder templateBuilder with options 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
    <head data-reactroot=\\"\\"><title>titleMock</title><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"bundle.css\\"/><meta charSet=\\"UTF-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"/><meta name=\\"metaMock\\" content=\\"metaContentMock\\"/></head>
    <body>
        <div id=\\"content\\"><div data-reactroot=\\"\\"><nav class=\\"navbar navbar-default navbar-fixed-top\\"><div class=\\"container-fluid\\"><div class=\\"navbar-header\\"><a href=\\"\\" class=\\"navbar-brand\\">React-Bootstrap</a><button type=\\"button\\" class=\\"navbar-toggle collapsed\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button></div><div class=\\"navbar-collapse collapse\\"><ul class=\\"nav navbar-nav navbar-right\\"><li class=\\"dropdown\\"><a id=\\"basic-nav-dropdown\\" role=\\"button\\" class=\\"dropdown-toggle\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" href=\\"#\\">Categories<!-- --> <span class=\\"caret\\"></span></a><ul role=\\"menu\\" class=\\"dropdown-menu\\" aria-labelledby=\\"basic-nav-dropdown\\"><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories\\" role=\\"menuitem\\" tabIndex=\\"-1\\">All</a></li><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories?filter=urban\\" role=\\"menuitem\\" tabIndex=\\"-1\\">Urban</a></li><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories?filter=nature\\" role=\\"menuitem\\" tabIndex=\\"-1\\">Nature</a></li></ul></li><li role=\\"presentation\\" class=\\"\\"><a role=\\"link\\" href=\\"contact\\">Contact Us</a></li></ul></div></div></nav><section class=\\"main-content\\"><div></div></section><footer class=\\"footer\\"><p>© André Bonna</p></footer></div></div>
        <script type=\\"text/javascript\\">
            window.APP_PROPS = '{\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"images\\":[]},\\"_owner\\":null,\\"_store\\":{}},\\"categories\\":[{\\"name\\":\\"All\\",\\"href\\":\\"categories\\"},{\\"name\\":\\"Urban\\",\\"href\\":\\"categories?filter=urban\\"},{\\"name\\":\\"Nature\\",\\"href\\":\\"categories?filter=nature\\"}],\\"page\\":\\"Home\\"}';
        </script>
        <script src=\\"bundle.js\\"></script>
    </body>
</html>"
`;
